common/src/main/java/org/axonframework/common/ReflectionUtils.java

```diff
@@ -584,6 +589,99 @@ public static String fieldNameFromMember(@Nonnull Member member) {
+                                     && method.getParameterCount() == 0
+                                     && method.getReturnType().equals(field.getType()));
+    }
+
+    /**
+     * Collects all {@link Member}s of type {@link Field} and {@link Method} for given type in one single list.
+     * If the given type is a record, the record-style getters are deduplicated.
+     *
+     * @param type the type which fields and methods are collectd
+     * @return list of all found fields and messages
+     * @see #collectAnnotatedMethodsAndFields(Class, Predicate)
+     */
+    public static List<? extends Member> collectAnnotatedMethodsAndFields(Class<?> type) {
+        return collectAnnotatedMethodsAndFields(type, it -> true);
+    }
+
+    /**
+     * Collects all {@link Member}s of type {@link Field} and {@link Method} which match the given filter
+     * for a given type in one single list.
+     * If the given type is a record, the record-style getters are deduplicated.
+     *
+     * @param type the type which fields and methods are collectd
+     * @param filter a filter that only keeps matching members
+     * @return list of all found fields and messages
+     */
+    public static List<? extends Member> collectAnnotatedMethodsAndFields(Class<?> type,
+                                                                                   Predicate<Member> filter) {
+        List<Method> methods = stream(ReflectionUtils.methodsOf(type).spliterator(), false).toList();
+        var deduplicateFilter = deduplicateRecordFields(methods);
+
+        var fields = stream(ReflectionUtils.fieldsOf(type).spliterator(), false)
+                .filter(deduplicateFilter);
+
+        return Stream.concat(fields, methods.stream())
+                     .filter(filter)
+                     .toList();
+    }
+
+    /**
+     * Collects all concrete types from a sealed hierarchy if the given type is sealed. Returns an empty set if the type
+     * is not sealed.
+     *
+     * @param rootType The root type to scan for sealed hierarchy.
+     * @param <T>      The type parameter.
+     * @return A set of all concrete types in the hierarchy, or an empty set if the type is not sealed.
+     */
+    public static <T> Set<Class<? extends T>> collectSealedHierarchyIfSealed(@Nonnull Class<T> rootType) {
+        if (!rootType.isSealed()) {
+            return Set.of();
+        }
+        Set<Class<? extends T>> result = new HashSet<>();
+        collectSealedHierarchyRecursive(rootType, result);
+        return result;
+    }
+
+    /**
+     * Recursively collects all concrete types from a sealed hierarchy.
+     *
+     * @param type        The current type to scan.
+     * @param accumulator The set to accumulate concrete types into.
+     * @param <T>         The root type parameter.
+     */
+    @SuppressWarnings("unchecked")
+    private static <T> void collectSealedHierarchyRecursive(@Nonnull Class<T> type,
```

Comment:
Just being curious here, but does this have to be recursive? That would mean you could have a sealed interface that permits 2 types, of which one is a sealed interface itself with several options.. Wasn't aware you could do that. Is that a valid scenario? 

If it is, then I think we'd need a test for that!